include/zephyr/net/prometheus/counter.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_PROMETHEUS_COUNTER_H_
+#define ZEPHYR_INCLUDE_PROMETHEUS_COUNTER_H_
+
+/**
+ * @file
+ *
+ * @brief Prometheus counter APIs.
+ *
+ * @addtogroup prometheus
+ * @{
+ */
+
+#include <zephyr/sys/iterable_sections.h>
+#include <zephyr/net/prometheus/metric.h>
+
+/**
+ * @brief Type used to represent a Prometheus counter metric.
+ *
+ * * References
+ * * See https://prometheus.io/docs/concepts/metric_types/#counter
+ */
+struct prometheus_counter {
+	/** Base of the Prometheus counter metric */
+	struct prometheus_metric *base;
+	/** Value of the Prometheus counter metric */
+	unsigned long value;
```

Comment:
For an event happening every 10 millisecond, AFAIU, where `unsigned long` is 32-bit, this would overflow every 1.3 year? Someone migrating from a 64-bit (overflow every  to a 32-bit platform could get surprised.


`

Reply:
Maybe this is not a matter to solve by this library but by the user?

If wishing to address this, using `uint64_t` or `uint32_t` (and documenting it) could help eventually.
With `uint32_t`, checking for overflow and reporting it could be good addition.

Not sure `uint64_t` will ever overflow if incrementing by just 1 every time:
```python
>>> 0xffffffffffffffff /  (1000000000 * 3600 * 24 * 365)
overflow every 584.942417355072 years if logging every nanosecond
```

Reply:
https://en.cppreference.com/w/cpp/language/types